windows-agent/internal/config/config.go

```diff
@@ -301,3 +311,49 @@ func hasChanged(newValue string, checksum *string) bool {
 	*checksum = newCheckSum
 	return true
 }
+
+// completeLandscapeConfig completes the Landscape configuration by adding the hostagent_uid field to the client section,
+// making it ready for consumption by the Landscape client inside the distro instances.
+func completeLandscapeConfig(landscapeConf, hostAgentUID string) (string, error) {
+	if landscapeConf == "" {
+		return "", nil
+	}
+	conf, err := ini.Load(strings.NewReader(landscapeConf))
+	if err != nil {
+		return "", fmt.Errorf("could not parse Landscape configuration: %v", err)
+	}
+
+	clientSection, err := conf.GetSection("client")
+	if err != nil {
+		return "", fmt.Errorf("could not find the client section in the Landscape configuration: %v", err)
+	}
+
+	if hostAgentUID == "" {
+		return landscapeConf, nil
+	}
+
+	keyName := "hostagent_uid"
+	if key, err := clientSection.GetKey(keyName); err == nil {
+		key.SetValue(hostAgentUID)
+	} else {
+		_, err = clientSection.NewKey(keyName, hostAgentUID)
+		if err != nil {
+			return "", fmt.Errorf("could not add the %s key to the client section: %v", keyName, err)
+		}
+	}
+
+	// Write the ini to a string
+	var b strings.Builder
+
+	// We're writing to machines, don't need to be pretty.
+	prettyFormat := ini.PrettyFormat
```

Comment:
Precisely to avoid diffs due spaces only. I don't like that either and wonder why a global for such configuration instead of an option at write time. I'll improve the comment.